File: frontend/src/api/firstRun.ts
import type { FirstRunActionResult, FirstRunReport, FirstRunStatus } from '../types/firstRun'

function apiBaseUrl(): string {
  const configured = import.meta.env.VITE_BACKEND_API_URL as string | undefined
  if (configured) return configured.replace(/\/$/, '')
  if (window.location.port && window.location.port !== '5173') return window.location.origin
  return `${window.location.protocol}//${window.location.hostname}:8000`
}

async function request<T>(path: string, init?: RequestInit): Promise<T> {
  const response = await fetch(`${apiBaseUrl()}${path}`, {
    headers: {
      'Content-Type': 'application/json',
      ...init?.headers,
    },
    ...init,
  })
  const body = await response.json()
  if (!response.ok) throw new Error(JSON.stringify(body))
  return body as T
}

export function fetchFirstRunStatus(): Promise<FirstRunStatus> {
  return request<FirstRunStatus>('/api/first-run/status')
}

export function runFirstRunCheck(): Promise<FirstRunReport> {
  return request<FirstRunReport>('/api/first-run/check', { method: 'POST' })
}

export function markFirstRunComplete(): Promise<FirstRunActionResult> {
  return request<FirstRunActionResult>('/api/first-run/mark-complete', { method: 'POST' })
}

export function resetFirstRun(): Promise<FirstRunActionResult> {
  return request<FirstRunActionResult>('/api/first-run/reset', { method: 'POST' })
}

export function fetchFirstRunReport(): Promise<FirstRunReport> {
  return request<FirstRunReport>('/api/first-run/report')
}

File: backend/app/api/first_run.py
from fastapi import APIRouter, Depends

from app.api.deps import get_runtime
from app.schemas.first_run import FirstRunActionResult, FirstRunReport, FirstRunStatus
from app.services.runtime_state import RuntimeState

router = APIRouter(prefix="/api/first-run", tags=["first-run"])


@router.get("/status", response_model=FirstRunStatus)
def status(runtime: RuntimeState = Depends(get_runtime)) -> FirstRunStatus:
    return runtime.first_run.status(runtime)


@router.post("/check", response_model=FirstRunReport)
def check(runtime: RuntimeState = Depends(get_runtime)) -> FirstRunReport:
    return runtime.first_run.check(runtime)


@router.post("/mark-complete", response_model=FirstRunActionResult)
def mark_complete(runtime: RuntimeState = Depends(get_runtime)) -> FirstRunActionResult:
    return runtime.first_run.mark_complete(runtime)


@router.post("/reset", response_model=FirstRunActionResult)
def reset(runtime: RuntimeState = Depends(get_runtime)) -> FirstRunActionResult:
    return runtime.first_run.reset(runtime)


@router.get("/report", response_model=FirstRunReport)
def report(runtime: RuntimeState = Depends(get_runtime)) -> FirstRunReport:
    return runtime.first_run.report(runtime)
